test/node/bigint.test.ts

```diff
@@ -0,0 +1,89 @@
+import { BSON } from '../register-bson';
+import { bufferFromHexArray } from './tools/utils';
+import { BSON_DATA_LONG } from '../../src/constants';
+
+describe('BSON BigInt serialization Support', function () {
+  // Index for the data type byte of a BSON document with a single element
+  const DATA_TYPE_OFFSET = 4;
+
+  it('Serializes bigints with the correct BSON type', function () {
+    const testDoc = { a: 0n };
+    const serializedDoc = BSON.serialize(testDoc);
+    expect(serializedDoc[DATA_TYPE_OFFSET]).to.equal(BSON_DATA_LONG);
+  });
+
+  it('Serializes bigints into little-endian byte order', function () {
+    const testDoc = { a: 0x1234567812345678n };
+    const serializedDoc = BSON.serialize(testDoc);
+    const expectedResult = bufferFromHexArray([
+      '12', // int64 type
+      '6100', // 'a' key with key null terminator
+      '7856341278563412'
+    ]);
+    expect(serializedDoc).to.deep.equal(expectedResult);
```

Comment:
Since we know the `DATA_TYPE_OFFSET` is always consistent, then the offset of the actual int is consistent too, can we assert that if we put the `expectedResult` in a DataView and use getBigInt64 with the LE flag set to true that it equals the input?